pkg/git/repository/connectivity.go

```diff
@@ -0,0 +1,71 @@
+package repository
+
+import (
+	"fmt"
+	"github.com/redhat-developer/devconsole-git/pkg/log"
+	gittransport "gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	unableReachRepoError   = "unable to reach the repo"
+	unableFindBranchError  = "unable to find the branch"
+	unableParseBranchError = "unable to parse the branch"
+)
+
+func IsReachableWithBranch(log *log.GitSourceLogger, branch string, endpoint *gittransport.Endpoint) (bool, error) {
+	if endpoint.Host == "" {
+		return false, fmt.Errorf(unableReachRepoError)
+	}
+	client := &http.Client{}
+	path := endpoint.Path
+	if !strings.HasSuffix(path, ".git") {
+		path += ".git"
+	}
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+	url := fmt.Sprintf("https://%s%s/info/refs?service=git-upload-pack", endpoint.Host, path)
+	resp, err := client.Get(url)
+	if err != nil {
+		return false, fmt.Errorf(unableReachRepoError)
+	} else {
```

Comment:
can we remove this else clause and just have 

`return validateBranch(log, branch, resp)`

Reply:
done 227e92b